lib/runner/cli/cli.js

```diff
@@ -485,6 +490,16 @@ class CliRunner {
         }
 
         return promise.then(() => {
+          if (this.sequentialMode()) {
+            return this.testEnvArray.reduce((prev, env) => {
+              return prev.then(async () => {
+                this.processListener.testRunner.settings = this.__testEnvSettings[env];
```

Comment:
should we use `this.testEnvSettings` over here?

Reply:
I agree with @gravityvi  and also there are too many nested callbacks here, move this to a different function inside `concurrency/index.js`

Reply:
Fixed too many callbacks and `__testEnvSettings`. Also if we move these changes to `concurrency/index.js` we have to pass `executeTestRunner` function which doesn’t looks good to me. Still I am open for suggestions.